src/core/or/relay.c

```diff
@@ -2537,6 +2538,7 @@ cell_queues_check_size(void)
   const size_t geoip_client_cache_total =
     geoip_client_cache_total_allocation();
   alloc += geoip_client_cache_total;
+  const size_t dns_cache_total = dns_cache_total_allocation();
```

Comment:
At this point, shouldn't we also add dns_cache_total to alloc? 

"alloc" is the total amount of bytes allocated that we might be able to free, so I think we should include the DNS bytes here, especially since we subtract them from "alloc" below.